### ISSUE TITLE
subsubsection and beyond text rendered inline on LaTeX

### ISSUE THREAD

**Issue by mcondarelli** [doxygen/doxygen#9368]
**Describe the bug**
Sections initiated with three hash or more (`###...`) have following text on the same line as title in LaTeX/PDF while they are rendered "correctly" (i.e.: on new line) in HTML rendering.

**Expected behavior**
Consistent behavior.

**Screenshots**
![image](https://user-images.githubusercontent.com/870849/171991489-06e87bf6-b670-416b-ab38-fcdd34cf1b87.png)

**To Reproduce**
Attached self contained example reproduces the problem.
[example.tar.gz](https://github.com/doxygen/doxygen/files/8837299/example.tar.gz)

**Version**
This happens both on 1.9.1 (Debian Sid standard) and in 1.9.5 (git HEAD)

**Comment by albert-github** [comment]
Difficult problem, I've posted in the tex part of stack exchange a question: https://tex.stackexchange.com/questions/646645/text-after-a-paragraph-heading-should-not-be-on-same-line

**Comment by albert-github** [comment]
The question at stack exchange gave some (hard to understand remarks) but based on it I was fortunate to create a proposed solution.

I've just pushed a proposed patch, pull request #9371

**Comment by mcondarelli** [comment]
Thanks @albert-github,
unfortunately this does **not** seem to work for me.

Construct like `###Post-processing:\nother stuff` result in truncation (till snippet-end)

Construct like `#Post-processing:\nother stuff` result as-is inclusion and the following warning:
```
CommandHandler.c:285: warning: explicit link request to 'Post' could not be resolved
```
I will keep on using `<h3>` ;)

**Comment by albert-github** [comment]
I think the syntax used is not correct,
The line:
```
###Post-processing:\nother stuff result in truncation (till snippet-end)
```
should read
```
### Post-processing:\nother stuff result in truncation (till snippet-end)
```

so with a space.

**Comment by mcondarelli** [comment]
Snippet
```
...
<h4>Post-processing:</h4>
Check if in previous loops we handled LRM command (if any), if not handle it.

<h4>General scan flowchart is as follows:</h4>
some visible text

# General scan flowchart is as follows:

some not visible text
...
```
results in:
![image](https://user-images.githubusercontent.com/870849/172004081-28beba97-aa02-4f27-ba4a-aaae46b4e808.png)

**Comment by albert-github** [comment]
This looks like to work for me though:
- What do you expect here?
- What is the complete example?

**Comment by mcondarelli** [comment]
I expected to see **two** "General scan flowchart is as follows:" sections: one generated by `<h4>` and one generated by `# `.

The second one, along with "some not visible text" (along with everything after it) is not anywhere (Chapter 6 is "Class Index" and has nothing to do with the above snippet)

This is part of my real doc, so I cannot send it, but I'll try to produce a minimal example (and I will also cross-check I didn't do anything stupid wit Doxygen recompilation).

Sorry being pestering you :( please forgive me ;)

**Comment by albert-github** [comment]
I retried with the current master (1.9.5 (410e8c5e60f9bfbb3be215167803bf8899e1f371)) and it looks OK to me. I see:

![image](https://user-images.githubusercontent.com/5223533/172012870-a88e70d1-f68c-42de-8f9a-ec92034f6cf6.png)

Awaiting the example.

**Comment by mcondarelli** [comment]
Then I **must** have goofed somewhere in building.

I currently have:
```
mcon@ikea:/usr/local/src/doxygen$ git log -1
commit 563918587e8cf90f54608768026a40b3786d00f5 (HEAD -> master, origin/master, origin/HEAD, tracking_albert)
Merge: 33c72f00 a4565784
Author: Dimitri van Heesch <doxygen@gmail.com>
Date:   Sat Jun 4 20:42:36 2022 +0200

    Merge pull request #9360 from makelinux/legend-node
    
    refactor DotLegendGraph::computeTheGraph: consolidate common node attributes
mcon@ikea:/usr/local/src/doxygen$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   templates/latex/doxygen.sty

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	albert01.patch
	albert02.patch

no changes added to commit (use "git add" and/or "git commit -a")
mcon@ikea:/usr/local/src/doxygen$ git diff
diff --git a/templates/latex/doxygen.sty b/templates/latex/doxygen.sty
index bde67edb..f58ae161 100644
--- a/templates/latex/doxygen.sty
+++ b/templates/latex/doxygen.sty
@@ -558,9 +558,9 @@
 % unfortunately had to copy part of book.cls and add \raggedright
 \makeatletter
 \newcommand\doxysection{\@startsection {section}{1}{\z@}%
-                                   {-3.5ex \@plus -1ex \@minus -.2ex}%
-                                   {2.3ex \@plus.2ex}%
-                                   {\raggedright\normalfont\Large\bfseries}}
+                                     {-3.5ex \@plus -1ex \@minus -.2ex}%
+                                     {2.3ex \@plus.2ex}%
+                                     {\raggedright\normalfont\Large\bfseries}}
 \newcommand\doxysubsection{\@startsection{subsection}{2}{\z@}%
                                      {-3.25ex\@plus -1ex \@minus -.2ex}%
                                      {1.5ex \@plus .2ex}%
@@ -570,13 +570,13 @@
                                      {1.5ex \@plus .2ex}%
                                      {\raggedright\normalfont\normalsize\bfseries}}
 \newcommand\doxyparagraph{\@startsection{paragraph}{4}{\z@}%
-                                    {3.25ex \@plus1ex \@minus.2ex}%
-                                    {-1em}%
-                                    {\raggedright\normalfont\normalsize\bfseries}}
+                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
+                                     {1.5ex \@plus .2ex}%
+                                     {\raggedright\normalfont\normalsize\bfseries}}
 \newcommand\doxysubparagraph{\@startsection{subparagraph}{5}{\parindent}%
-                                       {3.25ex \@plus1ex \@minus .2ex}%
-                                       {-1em}%
-                                      {\raggedright\normalfont\normalsize\bfseries}}
+                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
+                                     {1.5ex \@plus .2ex}%
+                                     {\raggedright\normalfont\normalsize\bfseries}}
 \makeatother
 % Define caption that is also suitable in a table
 \makeatletter
mcon@ikea:/usr/local/src/doxygen$ 
```
and build with:
```
mkdir build && cd build
cmake -G "Unix Makefiles" ..
make && sudo make install && sudo chown -Rc mcon.mcon .
```

Minimal Test program and results are in:
[example.tar.gz](https://github.com/doxygen/doxygen/files/8839659/example.tar.gz)

Please don't spend too much time on this.
As said I can easily work around using `<h#> tags.
This is a very minor hassle, but I would like to understand why I get different results :(

Many Thanks in Advance

**Comment by albert-github** [comment]
When I run the given example from https://github.com/doxygen/doxygen/issues/9368#issuecomment-1146771435 with the current master (1.9.5 (3f638177ce31c5d727b87e45f721bf9bd830c189))  and the `doxygen.sty` patch I get the correct output.

I don't know what is in the patches `albert01` and `albert02`  but I assume the changes regarding the `PROJECT_NUMBER` (this has been integrated in master already) and the changes in `doxygen.sty`. 

The results:

![image](https://user-images.githubusercontent.com/5223533/172046792-d22773bf-3f0f-4543-a5ad-a407e35d90c2.png)

and

![image](https://user-images.githubusercontent.com/5223533/172046811-2658013b-0fc7-4101-931c-0a7f1429fadd.png)

**Comment by Klypto** [comment]
I am uncertain if this will address specific the problem referred to in #9365 as this seems to specify only headers.

Even using latest c345c22 build this morning I still see the following output:

![image](https://user-images.githubusercontent.com/3277627/188902256-b7163c1c-53c2-498f-8e58-686095eafd1a.png)
![image](https://user-images.githubusercontent.com/3277627/188902274-554e48c1-3b4f-48cc-88f0-b5e607ba2ee6.png)

**Comment by albert-github** [comment]
@Klypto 
Don't use images to show code, as this cannot be cut /paste.
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by Klypto** [comment]
Here is a basic example of the specific problem I was seeing / was described in the issue mentioned.
[DoxygenBugReport.zip](https://github.com/doxygen/doxygen/files/9508284/DoxygenBugReport.zip)

If you feel this needs a separate report and is not related to the original issue I can submit it under a new one instead.

**Comment by albert-github** [comment]
@Klypto 
The original problem here is about the representation in LaTeX and not about the merging of the text. This issue (#9368) is a spin-off from the issue #9365 and your problem / example could be posted there. The problem was already identified there as well in https://github.com/doxygen/doxygen/issues/9365#issuecomment-1142377259 as:
> - the problem is only in the HTML output
> - the problem comes from the HTML paragraph (`<p>`) handling in doxygen

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
